index.php

```diff
@@ -43,7 +43,9 @@
 
     set_config('prodbgcolour', $data->prodbgcolour, 'local_envbar');
     set_config('prodtextcolour', $data->prodtextcolour, 'local_envbar');
-    set_config('secretkey', $data->secretkey, 'local_envbar');
+    if (!envbarlib::is_secret_key_overridden()) {
+        set_config('secretkey', $data->secretkey, 'local_envbar');
+    }
```

Comment:
should we reset the stored secretkey if overriden to avoid the old value staying there unused?

Reply:
I don't think so. But ready to discuss.